src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/RisksAndNeedsService.kt

```diff
@@ -92,22 +93,26 @@ class RisksAndNeedsService(
 
     // this endpoint requires user auth tokens for security reasons
     val authentication = SecurityContextHolder.getContext().authentication as JwtAuthenticationToken
-    return assessRisksAndNeedsClient.post(createSupplementaryRiskLocation, request, customAuthentication = authentication)
+    val response = assessRisksAndNeedsClient.post(createSupplementaryRiskLocation, request, customAuthentication = authentication)
       .retrieve()
-      .bodyToMono(SupplementaryRiskResponse::class.java)
-      .onErrorResume(WebClientResponseException::class.java) {
-        when (it.statusCode) {
-          HttpStatus.CONFLICT -> {
-            logger.warn(
-              "attempted to create new supplementary risk, but risk already exists for this referral {} {}",
-              kv("crn", crn),
-              kv("referralId", referralId)
-            )
-          }
-        }
-        Mono.error(it)
-      }
+      .onStatus({ it == HttpStatus.CONFLICT }, { Mono.empty() })
+      .toEntity(SupplementaryRiskResponse::class.java)
       .block()
-      .supplementaryRiskId
+
+    if (response.statusCode.equals(HttpStatus.CONFLICT)) {
+      if (response.body.createdDate != riskCreatedAt) {
+        logger.error(
+          "attempted to update an existing supplementary risk with new data {} {} {} {}",
+          kv("crn", crn),
+          kv("referralId", referralId),
+          kv("riskId", response.body.supplementaryRiskId),
+          kv("riskCreatedAt", riskCreatedAt)
+        )
+
+        throw WebClientResponseException(409, "Conflict from ARN 'createSupplementaryRisk'", null, null, null)
```

Comment:
Do we expect the user to be able to resolve the conflict and resubmit the request in this case?

Reply:
no - there's nothing they can do. which is the situation we are in now, regardless of whether or not they have changed the information they are trying to submit.

Reply:
i think it's unlikely the risk information will have changed for most cases. so hoping that this unblocks most people with referrals stuck in this state.

Reply:
we will now get sentry alerts for these cases too